### PR TITLE
AccessControl: Move GetCacheKey to SignedInUser

### LINKED ISSUE #3107: Singlestat "Error: Multiple Series Error"
In version 2.0 i use single stat with 2 rows, because the value it's a percent of row #A.
In the Row A i got the access metrics and in the Row B i got error metrics, so using the function "asPercent" i got this percentage of error in the access;
But in the new version i receive the following error:

Multiple Series Error
Metric query returns 2 series. Single Stat Panel expects a single series.

Can you help me please;

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What this PR does / why we need it**:
Add GetCacheKey, HasUniqueId and IsApiKeyUser to SignedInUser.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/user/model.go [hunk 1]
```diff
@@ -257,6 +259,24 @@ func (u *SignedInUser) IsRealUser() bool {
 	return u.UserID != 0
```
Comment (Jguer) [inline pkg/services/user/model.go:259]:
```suggestion
	return u.UserID > 0
```

File: pkg/services/user/model.go [hunk 2]
```diff
@@ -257,6 +259,24 @@ func (u *SignedInUser) IsRealUser() bool {
...
 }
 
+func (u *SignedInUser) IsApiKeyUser() bool {
+	return u.ApiKeyID != 0
```
Comment (Jguer) [inline pkg/services/user/model.go:263]:
```suggestion
	return u.ApiKeyID > 0
```

We've had scenarios where this values was changed to -1 and that does not correspond to a real user or API Key

**Review by Jguer — CHANGES_REQUESTED** [review]
The new additions are good :+1: , I believe you're missing the call replacement in the context handler

**Review by Jguer — APPROVED** [review]
My bad, didn't notice this was one in a series of PRs

File: pkg/services/user/model.go (hunk 1 above)
Comment (mgyongyosi) [reply pkg/services/user/model.go]: Changed, thanks!

File: pkg/services/user/model.go (hunk 2 above)
Comment (mgyongyosi) [reply pkg/services/user/model.go]: Changed, thanks!

**Pull request merged**